src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -235,7 +235,15 @@ protected void FindChangeAddress(TransactionBuildContext context)
         {
             // Get an address to send the change to.
             context.ChangeAddress = this.walletManager.GetUnusedChangeAddress(new WalletAccountReference(context.AccountReference.WalletName, context.AccountReference.AccountName));
-            context.TransactionBuilder.SetChange(context.ChangeAddress.ScriptPubKey);
+
+            if (context.UseSegwitChangeAddress)
```

Comment:
Instead of introducing a flag to the context, would it not make sense to make `GetUnusedChangeAddress` a few lines above more intelligent?